compiler/front/scriptconfig.nim

```diff
@@ -254,6 +254,20 @@ proc runNimScript*(cache: IdentCache; scriptName: AbsoluteFile;
   var m = graph.makeModule(scriptName)
   incl(m.flags, sfMainModule)
   var vm = setupVM(m, cache, scriptName.string, graph, idgen)
+  let isDisallowDanger =
+    defined(nimsuggest) or graph.config.cmd == cmdCheck or
+    vmopsDanger notin graph.config.features
+  # the VM instance used for NimScript execution is also used for the
+  # compile-time evaluation, so we have to register the macro/compile-time
+  # ops on the instance. Combined with how callback registration works,
+  # this has the following consequences:
+  # - CTFE and macro evaluation happening during semantic analysis of the
+  #  NimScript file are run with NimScript privileges, e.g. file-system write
+  #  access
+  # - the ``vmopsDanger`` option has no effect on callbacks registered
+  #  during `setupVM`
+  # - NimScript has access to the macro/compile-time APIs
```

Comment:
yup, stuck with this until we clarify compile time vs runtime, vm vs nimscript, host vs target, etc... oh well, that's ok for now.

Reply:
Each NimScript execution uses a separate VM instance (since a separate `ModuleGraph` is used) and while that is quite a waste of resources, it at least means that the side-effects listed here don't carry over to normal compilation.